webpack/scenes/ActivationKeys/Details/components/EditModal.js

```diff
@@ -0,0 +1,201 @@
+  };
+
+  const onMinus = () => {
+    setMaxHostsValue(oldValue => (oldValue || 0) - 1);
+  };
+  const onChange = (event) => {
+    let newValue = (event.target.value === '' ? event.target.value : Math.round(+event.target.value));
+    if (newValue < 1 && newValue !== '') {
+      newValue = 1;
+    }
+    setMaxHostsValue(newValue);
+  };
+  const onPlus = () => {
+    setMaxHostsValue(oldValue => (oldValue || 0) + 1);
+  };
+
+  const handleCheckBox = () => {
+    setUnlimited(prevUnlimited => !prevUnlimited);
+    setMaxHostsValue(usageCount > 0 ? usageCount : usageCount + 1);
+  };
+
+  return (
+    <>
+      <Button ouiaId="ak-edit-button" aria-label="edit-button" variant="secondary" onClick={handleModalToggle}>
+        {__('Edit')}
+      </Button>
+      <Modal
+        ouiaId="ak-edit-modal"
+        variant={ModalVariant.small}
+        title={__('Edit activation key')}
+        description={__(`Select attributes for ${akDetails.name}`)}
+        isOpen={isModalOpen}
+        onClose={handleClose}
+        actions={[
+          <Button ouiaId="edit-modal-save-button" key="create" variant="primary" form="modal-with-form-form" onClick={handleSave}>
+            {__('Save')}
+          </Button>,
+          <Button ouiaId="cancel-button" key="cancel" variant="link" onClick={handleClose}>
+            {__('Cancel')}
+          </Button>,
+        ]}
+      >
+        <Form isHorizontal>
+          <FormGroup
+            label={__('Name')}
+          >
+            <TextInput
+              ouiaId="ak-name-input"
+              id="ak-name-input"
+              type="text"
+              value={nameValue}
+              onChange={handleNameInputChange}
+            />
+          </FormGroup>
+          <FormGroup
+            label={__('Host Limit')}
+          >
+            <Stack hasGutter>
+              <StackItem>
+                <NumberInput
+                  value={maxHostsValue}
+                  min={1}
+                  onMinus={onMinus}
+                  onChange={onChange}
+                  onPlus={onPlus}
+                  inputName="input"
+                  inputAriaLabel="number input"
+                  minusBtnAriaLabel="minus"
+                  plusBtnAriaLabel="plus"
+                  isDisabled={isUnlimited}
+                  allowEmptyInput
+                />
+              </StackItem>
+              <StackItem>
+                <Checkbox
+                  ouiaId="unlimited-checkbox"
+                  id="unlimited-checkbox"
+                  label={__('Unlimited')}
+                  isChecked={isUnlimited}
+                  onChange={handleCheckBox}
+                />
+              </StackItem>
+            </Stack>
+          </FormGroup>
+          <FormGroup
+            label={__('Description')}
+          >
+            <TextArea
+              id="ak-description"
+              type="text"
+              placeholder={__('Description')}
+              value={descriptionValue}
```

Comment:
```suggestion
              value={descriptionValue || ''}
```
Console still complains about nil value for textarea field in edit modal if AK description is nil.